src/borg/archive.py

```diff
@@ -91,6 +91,7 @@ def show_progress(self, item=None, final=False, stream=None, dt=None):
                 space = columns - swidth(msg)
                 if space < swidth('...') + swidth(path):
                     path = '%s...%s' % (path[:(space // 2) - swidth('...')], path[-space // 2:])
+                space -= swidth(path)
                 msg += "{0:<{space}}".format(path, space=space)
```

Comment:
Yes, that would be clearer and it would be correct also.

I still think the current change is not correct. The resulting operation of your example is:

"{0:<4}".format(path_with_8_double_width_chars)

That operation is erroneous as it is trying to left-align a long string within a shorter field, but Python will still show you the whole string I guess as it never suppresses information.